# Analysis of Burke Gilman Trail Data
Mark 9/30 Big Data

## Abstract
In this lab we will be analyzing the Burke Gilman Trail data that is being published from February 4th 2014 to present. The questions that we will be trying to answer are: What is the average/highest/lowest amount of pedestrians to use this trail? What is the average/highest/lowest amount of bikers to use this trail?

## Dataset Preparation
Our dataset will be extracting:
- Pedestrians going north/south
- Bikers going north/south

We got our dataset from: https://data.seattle.gov/Transportation/Burke-Gilman-Trail-north-of-NE-70th-St-Bike-and-Pe/2z5v-ecg8

In [56]:
# Opening the file and storing all data in a list
file_data = open('Burke_Gilman_Trail_north_of_NE_70th_St_Bike_and_Ped_Counter.csv', 'r');
all_data = [];
for row in file_data:
    all_data.append(row);
file_data.close;


# Deleting the first row of data
del all_data[0];
# Adding all pedestrian data to correct list and all bike data to correct list
tot_ped = [];
tot_bike = [];
# Adding all pedestrian and bike data to the correct lists and cleaning the data of all empty or \n strings
for item in range (len(all_data)):
    data_split = all_data[item].split(",");
    if ((data_split[2] != '') and (data_split[2] != '\n')) :
        tot_ped.append(int(data_split[2]));
    if ((data_split[3] != '') and (data_split[3] != '\n')):
        tot_ped.append(int(data_split[3]));
    if ((data_split[4] != '') and (data_split[4] != '\n')):
        tot_bike.append(int(data_split[4]));
    if ((data_split[5] != '') and (data_split[5] != '\n')):
        tot_bike.append(int(data_split[5]));

I extracted all the data we are using into lists in this code. All the if statements that are run clean the dataset of all data in the set that is empty

## Data Modeling

In [57]:
# Getting the minumum and maximum amount of pedestrians and bikers
min_ped = min(tot_ped);
max_ped = max(tot_ped);
min_bike = min(tot_bike);
max_bike = max(tot_bike);

# Finding average amount of pedestrians and bikers
avg_ped = round(sum(tot_ped) / len(tot_ped));
avg_bike = round(sum(tot_bike) / len(tot_ped));


print("PEDESTRIAN DATA:")
print("The minimum number of pedestrians to go on the trail in an hour was", min_ped, "pedestrians");
print("The maximum number of pedestrians to go on the trail in an hour was", max_ped, "pedestrians");
print("The average number of pedestrians to go on the trail in an hour was", avg_ped, "pedestrians", '\n');
print("BIKE DATA");
print("The minimum number of bikers to go on the trail in an hour was", min_bike, "bikers");
print("The maximum number of bikers to go on the trail in an hour was", max_bike, "bikers");
print("The average number of bikers to go on the trail in an hour was", avg_bike, "bikers");

PEDESTRIAN DATA:
The minimum number of pedestrians to go on the trail in an hour was 0 pedestrians
The maximum number of pedestrians to go on the trail in an hour was 4095 pedestrians
The average number of pedestrians to go on the trail in an hour was 15 pedestrians 

BIKE DATA
The minimum number of bikers to go on the trail in an hour was 0 bikers
The maximum number of bikers to go on the trail in an hour was 8191 bikers
The average number of bikers to go on the trail in an hour was 22 bikers


To find all of the data for the pedestrians and bikers, I used the python keywords `min`, `max`, `sum` and `len` to find the minimum, maximum and average biker/pedestrian count. To find the average, I added up the total count of the list using `sum` then divided that by `len`.

## Data Analysis and Conclusion

Using this data, I found that the average amount of bikers in an hour was 22 bikers and the average number of pedestrians in an hour was 15. I also learned that the lowest number of pedestrians to use the trail in an hour was 0 and the highest was 4095. The lowest number of bikers to use the trail in an hour was 0 and the highest number of them was 8191. This brings me to the conclusion that bike and pedestrian usage varies greatly depending on the hour and day. Sometimes a trail can get 0 bikers or pedestrians in an hour and some other time it can get thousands of bikers/pedestrians in an hour
Some questions I have are: 
- Why did 8191 bikers go on the trail in an hour? Was there a special event?
- Why did 4095 pedestrians go on the trail in an hour? Was there some special
- What is the most common time for there to be 0 bikers/pedestrians on a trail in an hour?

## Acknowlegements
- Ms. Sconyers for helping me solve some coding problems I had
- Rafael Zuniga for creating the dataset I used